source/common/http/filter_manager.cc

```diff
@@ -625,6 +649,7 @@ void FilterManager::decodeData(ActiveStreamDecoderFilter* filter, Buffer::Instan
   // Filter iteration may start at the current filter.
   std::list<ActiveStreamDecoderFilterPtr>::iterator entry =
       commonDecodePrefix(filter, filter_iteration_start_state);
+  bool terminal_filter_decoded_end_stream = false;
```

Comment:
should we be aborting or asserting in `FilterManager::decode*` that the `decoder_filter_chain_complete_` is the appropriate value (false for assert, true if aborting)

Reply:
Adding ASSERTs revelead pre-existing bug in the deferred processing. I'm creating a PR to fix it.

Reply:
I have added ASSERTs